app/src/main/java/org/oppia/android/app/databinding/ConstraintViewAdapters.java

```diff
@@ -0,0 +1,42 @@
+package org.oppia.android.app.databinding;
+
+import android.view.View;
+
+import androidx.annotation.NonNull;
+import androidx.constraintlayout.widget.ConstraintLayout;
+import androidx.constraintlayout.widget.ConstraintSet;
+import androidx.databinding.BindingAdapter;
+
+/** Holds all custom binding adapters that set constraints to views. */
+public final class ConstraintViewAdapters {
+
+  /** Binding adapter for setting a [layout_constraintEndToEndOf] to a view */
+  @BindingAdapter("app:layout_constraintEnd_toEndOf")
+  public static void setConstraintEndToEndOf(
+      @NonNull View view, int constraintToId
+  ) {
+    ConstraintLayout constraintLayout = (ConstraintLayout) view.getParent();
+    ConstraintSet constraintSet = new ConstraintSet();
+    constraintSet.clone(constraintLayout);
+    constraintSet.connect(view.getId(), ConstraintSet.END, constraintToId, ConstraintSet.END);
+    constraintSet.applyTo(constraintLayout);
+  }
+
+  /** Binding adapter for setting a [layout_constraintStartToStartOf] to a view */
+  @BindingAdapter("app:layout_constraintStart_toStartOf")
+  public static void setConstraintStartToStartOf(
+      @NonNull View view, boolean setStartConstraint
```

Comment:
This confuses me--we should keep the same contract as the adapter above and bind to a specific ID rather than a boolean. ConstraintLayout doesn't ever bind to booleans, only ever to view IDs.

Reply:
@BenHenning Here, we want to constraint it with parent if ```viewModel.hasConversationView``` is ```false``` and we want to clear the constraint if it is ```true```, so what id should be passed if the condition is true?
Should we pass the id of the view itself, so we can check -
``` 
if (id==view.getid() {
    clear constraints
} else {
    connect to id 
}
```

Reply:
@prayutsu, in this case, I think you don't have to override `app:layout_constraintStart_toStartOf` itself. Instead, I suggest renaming it to something else, `shouldConnectToParent` for example.

Reply:
@MohamedMedhat1998 Definitely, if we finalize to use the mentioned approach, then I'll rename it as suggested by you.

Reply:
@BenHenning Can you please finalize the approach here?

Reply:
Out of curiosity, could we also always set the start constraint to the parent and then conditionally change the horizontal bias based on whether it's in conversation view? E.g a horizontal bias of 0 would be for the question player to left-align, and a bias of 1 would be used for the conversation view to right-align. Would that work?

Reply:
> Out of curiosity, could we also always set the start constraint to the parent and then conditionally change the horizontal bias based on whether it's in conversation view? E.g a horizontal bias of 0 would be for the question player to left-align, and a bias of 1 would be used for the conversation view to right-align. Would that work?

@BenHenning I think that should work, should I try it then?

Reply:
@prayutsu yes, I think that will be simpler since it avoids redefining the custom adapter in the way you did earlier, and avoids the need for a  completely custom property that's application-specific. I know it still probably requires a custom binding adapter for horizontal bias, but it's at least not redefinining how horizontal bias works.